draft-thomson-http-oblivious.md

```diff
@@ -916,6 +916,50 @@ corresponds to the key identifier, but the encapsulated request cannot be
 successfully decrypted using the key.
 
 
+## Replay Attacks
+
+Encapsulated requests can be copied and replayed toward servers. The replay
+risk is similar to the exposure to replay attack in TLS early data (see
+{{!TLS=RFC8446}}).
+
+A client or oblivious proxy resource MUST NOT automatically attempt to retry a
+failed request unless it receives a positive signal indicating that the request
+was not processed or forwarded. The HTTP/2 REFUSED_STREAM error code (Section
+8.1.4 of {{!RFC7540}}), the HTTP/3 H3_REQUEST_REJECTED error code (Section 8.1
+of {{!QUIC-HTTP=I-D.ietf-quic-http}}) and a GOAWAY frame (in either protocol
+version) are all sufficient signals that no processing occurred.
```

Comment:
```suggestion
version) are all sufficient signals that no processing occurred. Connection failures or interruptions
are not sufficient signals that no processing occurred. 
```

The client's connection to the proxy could idle timeout, at which point the client's signal is lost.